From 74e2e79c7e212ff9803a30a18c88bdbbee314800 Mon Sep 17 00:00:00 2001
From: Ian Dannapel <ian.dannapel@iris-sensing.com>
Date: Fri, 25 Mar 2022 17:25:28 +0100
Subject: [PATCH] i2c-imx: remove unneeded sleep
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

a delay on the startup of the i2c imx controller is not required or defined in the specs.
By removing it, the user can see a latency decrease from up to 150Î¼s in communication.

more info: http://patchwork.ozlabs.org/project/linux-i2c/patch/20220314095918.50014-1-iansdannapel@gmail.com/

Signed-off-by: Ian Dannapel <ian.dannapel@iris-sensing.com>
---
 drivers/i2c/busses/i2c-imx.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/drivers/i2c/busses/i2c-imx.c b/drivers/i2c/busses/i2c-imx.c
index b68068670adb..a34f16b7e831 100644
--- a/drivers/i2c/busses/i2c-imx.c
+++ b/drivers/i2c/busses/i2c-imx.c
@@ -691,12 +691,6 @@ static int i2c_imx_start(struct imx_i2c_struct *i2c_imx, bool atomic)
 	imx_i2c_write_reg(i2c_imx->hwdata->i2sr_clr_opcode, i2c_imx, IMX_I2C_I2SR);
 	imx_i2c_write_reg(i2c_imx->hwdata->i2cr_ien_opcode, i2c_imx, IMX_I2C_I2CR);
 
-	/* Wait controller to be stable */
-	if (atomic)
-		udelay(50);
-	else
-		usleep_range(50, 150);
-
 	/* Start I2C transaction */
 	temp = imx_i2c_read_reg(i2c_imx, IMX_I2C_I2CR);
 	temp |= I2CR_MSTA;
-- 
2.39.1

